windows/installer.nsi

```diff
@@ -365,45 +379,81 @@ Function createdesktopshortcut
    CreateShortCut  "$DESKTOP\${APP_NAME}.lnk" "$INSTDIR\${APP_EXE}"
 FunctionEnd
 
-Function ASIOCheckInstalled
-
-    ; insert ASIO install page if no ASIO driver was found
-    ClearErrors
-    EnumRegKey $0 HKLM "SOFTWARE\ASIO" 0
-
-    IfErrors 0 ASIOExists
-        !insertmacro MUI_HEADER_TEXT "$(ASIO_DRIVER_HEADER)" "$(ASIO_DRIVER_SUB)"
-        nsDialogs::Create 1018
-        Pop $Dialog
-        ${If} $Dialog == error
-            Abort
-        ${Endif}
-
-        ${NSD_CreateLabel} 0 0 100% 20u "$(ASIO_DRIVER_EXPLAIN)"
-        Pop $Label
-        ${NSD_CreateButton} 0 21u 100% 15u "$(ASIO_DRIVER_MORE_INFO)"
-        Pop $Button
-        ${NSD_OnClick} $Button OpenASIOHelpPage
-
-        nsDialogs::Show
-
-    ASIOExists:
-
-FunctionEnd
-
-Function OpenASIOHelpPage
-    ExecShell "open" "$(ASIO_DRIVER_MORE_INFO_URL)"
-FunctionEnd
-
-Function ExitASIOInstalled
-    ClearErrors
-    EnumRegKey $0 HKLM "SOFTWARE\ASIO" 0
-    IfErrors 0 SkipMessage
-        MessageBox MB_YESNO|MB_ICONEXCLAMATION "$(ASIO_EXIT_NO_DRIVER)" /sd IDNO IDYES SkipMessage
-            Abort
-   SkipMessage:
-
-FunctionEnd
+; Define functions for JACK when using the jackonwindows buildoption, else use default ASIO functions
+!if ${BUILD_OPTION} == "jackonwindows"
+    Function JACKCheckInstalled
+
+        ; insert JACK install page if no JACK driver was found
+        ClearErrors
+        EnumRegKey $0 HKLM "SOFTWARE\JACK" 0
+
+        IfErrors 0 JACKExists
+            !insertmacro MUI_HEADER_TEXT "$(JACK_DRIVER_HEADER)" "$(JACK_DRIVER_SUB)"
+            nsDialogs::Create 1018
+            Pop $Dialog
+            ${If} $Dialog == error
+                Abort
+            ${Endif}
+
+            ${NSD_CreateLabel} 0 0 100% 40u "$(JACK_DRIVER_EXPLAIN)"
+            Pop $Label
+
+            nsDialogs::Show
+
+        JACKExists:
+
+    FunctionEnd
+
+    Function ExitJACKInstalled
+        ClearErrors
+        EnumRegKey $0 HKLM "SOFTWARE\JACK" 0
+        IfErrors 0 SkipMessage
+            MessageBox MB_YESNO|MB_ICONEXCLAMATION "$(JACK_EXIT_NO_DRIVER)" /sd IDNO IDYES SkipMessage
+                Abort
+       SkipMessage:
+
+    FunctionEnd
+!else
+    Function ASIOCheckInstalled
+
+        ; insert ASIO install page if no ASIO driver was found
+        ClearErrors
+        EnumRegKey $0 HKLM "SOFTWARE\ASIO" 0
+
+        IfErrors 0 ASIOExists
+            !insertmacro MUI_HEADER_TEXT "$(ASIO_DRIVER_HEADER)" "$(ASIO_DRIVER_SUB)"
+            nsDialogs::Create 1018
+            Pop $Dialog
+            ${If} $Dialog == error
+                Abort
+            ${Endif}
+
+            ${NSD_CreateLabel} 0 0 100% 20u "$(ASIO_DRIVER_EXPLAIN)"
+            Pop $Label
+            ${NSD_CreateButton} 0 21u 100% 15u "$(ASIO_DRIVER_MORE_INFO)"
+            Pop $Button
+            ${NSD_OnClick} $Button OpenASIOHelpPage
+
+            nsDialogs::Show
+
+        ASIOExists:
+
+    FunctionEnd
```

Comment:
Does NSIS allow returning bools from functions? If so, it would be great to have something like 

if (apiRegKeyExists()) {
...
} else {
  throw error message.
}

Reply:
You have "then" and "else" targets, from what I remember.

Reply:
https://nsis.sourceforge.io/Macro_vs_Function#Hybrid

Something like
* Push the return value onto the stack
* Pop it in the caller
* If test to switch between the targets

Reply:
I would say this does not make it clearer to me, just adds extra complexity to the code.